crates/sui-indexer/src/store/pg_indexer_store.rs

```diff
@@ -1295,6 +1328,84 @@ impl PgIndexerStore {
         Ok(())
     }
 
+    async fn update_watermarks(&self, watermarks: Vec<Watermark>) -> Result<(), IndexerError> {
```

Comment:
Given that each kind of `Watermark` needs to be handled differently, and it seems like we only ever perform a batch of one kind of update, does it make sense to split these up into three separate functions?

Reply:
gah ... i actually started off as three separate functions, and then consolidated them into one :sweat_smile:

was thinking maybe later on we'd have a standalone watermark task that gets fed batches of updates, and having one entrypoint would be easier than three. but i can go either way

Reply:
I suspect you won't be able to introduce a completely standalone watermark task, because some of the watermark updates will need to be atomic with the change that affected the watermark (the changes that are used for pruner disaster recovery).

Even without that detail though, I would lean towards structuring the code in the way that is simplest for your immediate needs, and you can add the abstraction you're talking about when you need it.